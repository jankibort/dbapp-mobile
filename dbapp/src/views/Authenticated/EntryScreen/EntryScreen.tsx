import React, { FC, useContext, useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Button, ControlledInput } from '../../../components';
import { COLORS } from '../../../constant';
import { format } from 'date-fns';
import { SugarContext } from '../../../context';
import { useToast } from 'react-native-toast-notifications';
import { SugarRangesType } from '../../../context/SugarContex';

const FormState = {
  insulinAmount: '',
  sugarLevel: '',
};

export const EntryScreen: FC = () => {
  const [formState, setFormState] = useState<typeof FormState>(FormState);
  const { setInsulinRecord, setSugarRecord } = useContext(SugarContext);

  const toast = useToast();

  const getSugarRange = (value: number): SugarRangesType => {
    switch (true) {
      case value < 90:
        return 'low';
      case value >= 90 && value < 140:
        return 'ok';
      case value >= 140:
        return 'high';
      default:
        return 'ok';
    }
  };

  const handleSubmit = () => {
    if (formState.insulinAmount || formState.sugarLevel) {
      const dateTime = new Date();

      formState.insulinAmount &&
        setInsulinRecord &&
        setInsulinRecord((records) => [
          ...records,
          {
            amount: Number(formState.insulinAmount),
            dateTime,
          },
        ]);

      formState.sugarLevel &&
        setSugarRecord &&
        setSugarRecord((records) => [
          ...records,
          {
            value: Number(formState.sugarLevel),
            range: getSugarRange(Number(formState.sugarLevel)),
            dateTime,
          },
        ]);

      toast.show(`Data succesfully saved`, {
        duration: 3000,
        animationType: 'zoom-in',
        type: 'success',
        placement: 'top',
      });

      setFormState(FormState);
    } else {
      toast.show(`Fill in the data`, {
        duration: 3000,
        animationType: 'zoom-in',
        type: 'warning',
        placement: 'top',
      });
    }
  };

  const handleInputChange = (
    value: string,
    target: 'insulinAmount' | 'sugarLevel',
  ) => {
    formState && setFormState({ ...formState, [target]: value });
  };

  return (
    <View style={styles.container}>
      <Text style={styles.label}>Injected insulin amount</Text>
      <ControlledInput
        value={formState?.insulinAmount || ''}
        onChange={(val) => handleInputChange(val, 'insulinAmount')}
      />
      <Text style={styles.label}>Sugar Level</Text>
      <ControlledInput
        value={formState?.sugarLevel || ''}
        onChange={(val) => handleInputChange(val, 'sugarLevel')}
      />
      <Button
        title="Add entry!"
        onPress={handleSubmit}
        buttonStyle={{
          backgroundColor: COLORS.SUCCESS,
          alignSelf: 'center',
          width: '100%',
          marginTop: 30,
        }}
        labelStyle={{ color: COLORS.LIGHT, fontSize: 20 }}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: { width: 280, flex: 1, alignSelf: 'center', marginTop: 20 },
  contentWrapper: {
    height: '100%',
    backgroundColor: COLORS.COMPONENT.NON_AUTH_BG,
  },
  label: {
    fontSize: 20,
    color: COLORS.PRIMARY,
  },
  buttonWrapper: {
    flex: 1,
    alignSelf: 'center',
    flexDirection: 'column',
  },
});
