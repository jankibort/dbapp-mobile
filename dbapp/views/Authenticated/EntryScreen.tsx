import React, { FC } from 'react';

type Props = {};

export const EntryScreen: FC<Props> = () => {
  return <div>NewRecordForm</div>;
};
