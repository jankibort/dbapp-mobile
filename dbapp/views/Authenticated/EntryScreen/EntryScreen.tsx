import React, { FC, useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { ControlledInput } from '../../../components';
import { COLORS } from '../../../constants';

const FormState = {
  insulinAmount: 0,
  date: new Date(),
  carbohydrates: 0,
};

export const EntryScreen: FC = () => {
  const [formState, setFormState] = useState<typeof FormState>(FormState);

  const handleInputChange = (
    value: string,
    target: 'insulinAmount' | 'date' | 'carbohydrates',
  ) => {
    setFormState({ ...formState, [target]: value });
  };

  return (
    <View>
      <Text>Add record</Text>
      <Text style={styles.label}>Insulin amount</Text>
      <ControlledInput
        value={formState.insulinAmount === 0 ? '' : formState.insulinAmount}
        onChange={(val) => handleInputChange(val, 'insulinAmount')}
      />
      <Text style={styles.label}>Insulin amount</Text>
      <ControlledInput
        value={String(formState.date)}
        onChange={(val) => handleInputChange(val, 'date')}
      />
      <Text style={styles.label}>Insulin amount</Text>
      <ControlledInput
        value={formState.insulinAmount === 0 ? '' : formState.insulinAmount}
        onChange={(val) => handleInputChange(val, 'insulinAmount')}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  contentWrapper: {
    height: '100%',
    backgroundColor: COLORS.COMPONENT.NON_AUTH_BG,
  },
  label: {
    fontSize: 20,
    color: COLORS.LIGHT,
  },
  buttonWrapper: {
    flex: 1,
    alignSelf: 'center',
    flexDirection: 'column',
  },
});
