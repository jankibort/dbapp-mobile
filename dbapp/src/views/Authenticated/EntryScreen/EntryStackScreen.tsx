import { useNavigation } from '@react-navigation/native';
import {
  createNativeStackNavigator,
  NativeStackNavigationProp,
} from '@react-navigation/native-stack';
import React, { FC, useEffect } from 'react';
import { EntryScreen } from './EntryScreen';

export type EntryStackParams = {
  Entry: undefined;
};

const EntryStack = createNativeStackNavigator<EntryStackParams>();

/**
 *
 * @returns stack component which is wrapper to create nested stack navigation insinde tab nav
 */
export const EntryStackScreen: FC = () => {
  const navigation =
    useNavigation<NativeStackNavigationProp<EntryStackParams>>();

  useEffect(() => {
    navigation.navigate('Entry');
  }, []);

  return (
    <EntryStack.Navigator
      initialRouteName="Entry"
      screenOptions={{ headerShown: false }}
    >
      <EntryStack.Screen
        name="Entry"
        component={EntryScreen}
        options={{ title: 'New entry' }}
      />
    </EntryStack.Navigator>
  );
};
