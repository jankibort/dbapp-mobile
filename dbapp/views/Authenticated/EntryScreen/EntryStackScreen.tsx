import { useNavigation } from '@react-navigation/native';
import {
  createNativeStackNavigator,
  NativeStackNavigationProp,
} from '@react-navigation/native-stack';
import React, { FC, useEffect } from 'react';
import { EntryScreen } from './EntryScreen';

export type EntryStackParams = {
  Entry: undefined;
};

const EntryStack = createNativeStackNavigator<EntryStackParams>();

export const EntryStackScreen: FC = () => {
  const navigation =
    useNavigation<NativeStackNavigationProp<EntryStackParams>>();

  useEffect(() => {
    navigation.navigate('Entry');
    console.log('navigated to entry');
  }, []);

  return (
    <EntryStack.Navigator
      initialRouteName="Entry"
      screenOptions={{ headerShown: false }}
    >
      <EntryStack.Screen name="Entry" component={EntryScreen} />
    </EntryStack.Navigator>
  );
};
