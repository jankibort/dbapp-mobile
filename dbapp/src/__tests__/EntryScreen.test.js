import React from 'react';
import renderer from 'react-test-renderer';
import { EntryScreen } from '../views/Authenticated/EntryScreen/EntryScreen';

test('renders correctly', () => {
  const tree = renderer.create(<EntryScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
