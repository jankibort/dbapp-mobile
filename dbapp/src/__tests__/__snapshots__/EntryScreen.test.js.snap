// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "flex": 1,
      "marginTop": 20,
      "width": 280,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#317cf5",
        "fontSize": 20,
      }
    }
  >
    Injected insulin amount
  </Text>
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderColor": "#317cf5",
          "borderRadius": 6,
          "borderWidth": 1,
          "height": 36,
          "marginVertical": 12,
          "padding": 10,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Object {
        "color": "#317cf5",
        "fontSize": 20,
      }
    }
  >
    Sugar Level
  </Text>
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "borderColor": "#317cf5",
          "borderRadius": 6,
          "borderWidth": 1,
          "height": 36,
          "marginVertical": 12,
          "padding": 10,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#198754",
          "marginTop": 30,
          "width": "100%",
        },
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderRadius": 6,
          "justifyContent": "space-around",
          "minHeight": 26,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
          "textAlignVertical": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 20,
          },
        ]
      }
    >
      Add entry!
    </Text>
  </View>
</View>
`;
